src/c1_state_machine/p5_digital_cash.rs

```diff
@@ -48,9 +99,197 @@ impl StateMachine for DigitalCashSystem {
+            receives: vec![
+                Bill { owner: User::Bob, amount: 1000, serial: 33 }
+            ]
+        }
+    );
+    let expected = State::from([
+        Bill { owner: User::Alice, amount: 32, serial: 0 },
+    ]);
+    assert_eq!(end, expected);
+}
+
+#[test]
+fn spending_from_alice_to_all() {
+    let start = State::from([
+        Bill { owner: User::Alice, amount: 42, serial: 0 }
+    ]);
+    let end = DigitalCashSystem::next_state(
+        &start,
+        &CashTransaction::Transfer {
+            spends: vec![
+                Bill { owner: User::Alice, amount: 42, serial: 0 }
+            ],
+            receives: vec![
+                Bill { owner: User::Alice, amount: 10, serial: 1 },
+                Bill { owner: User::Bob, amount: 10, serial: 2 },
+                Bill { owner: User::Charlie, amount: 10, serial: 3 }
+            ]
+        }
+    );
+    let expected = State::from([
+        Bill { owner: User::Alice, amount: 10, serial: 1 },
+        Bill { owner: User::Bob, amount: 10, serial: 2 },
+        Bill { owner: User::Charlie, amount: 10, serial: 3 }
+    ]);
+    assert_eq!(end, expected);
+}
+
+#[test]
+fn spending_from_bob_to_all() {
+    let start = State::from([
+        Bill { owner: User::Bob, amount: 42, serial: 0 }
+    ]);
+    let end = DigitalCashSystem::next_state(
+        &start,
+        &CashTransaction::Transfer {
+            spends: vec![
+                Bill { owner: User::Bob, amount: 42, serial: 0 },
+            ],
+            receives: vec![
+                Bill { owner: User::Alice, amount: 10, serial: 1 },
+                Bill { owner: User::Bob, amount: 10, serial: 2 },
+                Bill { owner: User::Charlie, amount: 22, serial: 3 }
+            ]
+        }
+    );
+    let expected = State::from([
+        Bill { owner: User::Alice, amount: 10, serial: 1 },
+        Bill { owner: User::Bob, amount: 10, serial: 2 },
+        Bill { owner: User::Charlie, amount: 22, serial: 3 }
+    ]);
+    assert_eq!(end, expected);
+}
+
+#[test]
+fn spending_from_charlie_to_all() {
+    let mut start = State::from([
+        Bill { owner: User::Charlie, amount: 68, serial: 54 },
+        Bill { owner: User::Alice, amount: 4000, serial: 58 }
+    ]);
+    start.set_serial(59);
+    let end = DigitalCashSystem::next_state(
+        &start,
+        &CashTransaction::Transfer {
+            spends: vec![
+                Bill { owner: User::Charlie, amount: 68, serial: 54 }
+            ],
+            receives: vec![
+                Bill { owner: User::Alice, amount: 42, serial: 59 },
+                Bill { owner: User::Bob, amount: 5, serial: 60 },
+                Bill { owner: User::Charlie, amount: 5, serial: 61 }
+            ]
+        }
+    );
+    let mut expected = State::from([
+        Bill { owner: User::Alice, amount: 42, serial: 59 },
+        Bill { owner: User::Bob, amount: 5, serial: 60 },
+        Bill { owner: User::Charlie, amount: 5, serial: 61 }
+    ]);
```

Comment:
```suggestion
        let mut expected = State::from([
        Bill { owner: User::Alice, amount: 4000, serial: 58 },
        Bill { owner: User::Alice, amount: 42, serial: 59 },
        Bill { owner: User::Bob, amount: 5, serial: 60 },
        Bill { owner: User::Charlie, amount: 5, serial: 61 }
        ]);
```
```

Reply:
Same confusion for me here

Reply:
This is a nice catch. This was a mistake. The 4000 Alice bill never got spent. Except the expected would be 4000 with a serial number of 54

Reply:
Actually this was a mistake nice catch!